modules/data.atmosphere/R/download.ECMWF.R

```diff
@@ -0,0 +1,242 @@
+  # }
+  
+  # for forecast hour = 00 
+  if (fchour == 00){
+    time <- 0
+  } else {
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid forecast hour. Currently, forecast hour 00 is the default."
+    ))
+  }
+  
+  ############## Product (Forecasting system) ################
+  
+  # For product = NULL,the stream is set to "enfo"
+  # stream is forecasting system that produces the data. Current value is `"enfo"`
+  
+  if(is.null(product)){
+    stream <- "enfo"
+  } else{
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid data stream. Currently, data stream `enfo` is supported."
+    ))
+  }
+  
+  if (stream == "enfo"){
+    time <- time
+    step_15day <- 360
+    type <- type # type of forecast here c("cf", "pf")
+  } else{
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid data stream. Currently, data stream `enfo` is supported."
+    ))
+  }
+  
+  
+  ############## Product Types ################
+  
+  # Currently supported parameters are
+  # "2t" - 2 metre temperature 
+  # "tp" - total precipitation
+  # "10u" - 10 metre U wind component
+  # "10v" - 10 metre V wind component
+  # "sp" - Surface Pressure
+  
+  
+  all_parameters <- c("2t", "tp", "10u", "10v", "sp")
+  
+  if (tolower(parameters) == "all") {
+    parameter_types <- all_parameters
+  }
+  
+  if (any(!parameter_types %in% all_parameters)) {
+    bad_parameters <- setdiff(parameter_types, all_parameters)
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid parameter types %s. Products must be one of the following: %s",
+      paste0("`", bad_parameters, "`", collapse = ", "),
+      paste0("`", all_parameters, "`", collapse = ", ")
+    ))
+  }
+  
+  
+  # Python script "download_ecmwf.py" to get latest forecast date and download forecast datasets
+  script.path = file.path(system.file("ECMWF/download_ecmwf.py", package = "PEcAn.data.atmosphere"))
+  reticulate::source_python(script.path)
+  
+  date_latestdata <- ecmwflatest(time, step, stream, type, all_parameters)
+  latest_filedate <- strtoi(gsub("-", "", substr(as.character.Date(date_latestdata), 1, 10)))
+  
+  current_date <- strtoi(gsub("-", "", Sys.Date()))
+  
+  date <- if (latest_filedate == current_date){
+    # today
+    0
+  } else if ((latest_filedate - current_date) == -1){
+    # yesterday
+    -1
+  } else if ((latest_filedate - current_date) == -2){
+    # the day before yesterday
+    -2
+  } else { PEcAn.logger::logger.severe(sprintf(
+    "Invalid file download date."
+  ))
+  }
+  
+  # Skip download if there is any already existing file with base name same as `latest_filedate``
+  if ((length(list.files(path = outfolder, pattern = as.character(latest_filedate)) ) > 0) == TRUE){
+    PEcAn.logger::logger.severe(sprintf(
+      "Files already exist for %s",
+      latest_filedate
+    ))
+  } else {
+    
+    in_fname <- paste(latest_filedate, time, stream, sep = "_")
+    
+    data_download <- ecmwfdownload(date, time, stream, type, all_parameters, in_fname)
```

Comment:
I haven't thoroughly tested, but this function seems to be downloading the grib files to the working directory whereas it probably should use `outfolder`